<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>RulerView</class>
 <widget class="QWidget" name="RulerView">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>848</width>
    <height>483</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>交互测量</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_3">
   <item row="1" column="1">
    <widget class="QFrame" name="frame_2">
     <property name="maximumSize">
      <size>
       <width>300</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="title">
         <string>输入</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <property name="topMargin">
          <number>2</number>
         </property>
         <property name="bottomMargin">
          <number>2</number>
         </property>
         <item row="1" column="2">
          <widget class="QPushButton" name="pushButton_selectOrigin">
           <property name="text">
            <string>原图</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0" rowspan="4" colspan="2">
          <widget class="QFrame" name="frame">
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Raised</enum>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout">
            <property name="topMargin">
             <number>2</number>
            </property>
            <property name="bottomMargin">
             <number>2</number>
            </property>
            <item>
             <widget class="QCheckBox" name="checkBox_RawDispOK">
              <property name="text">
               <string>原始视差文件</string>
              </property>
              <property name="checkable">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="checkBox_OriginOK">
              <property name="text">
               <string>原图</string>
              </property>
              <property name="checkable">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="checkBox_CameraParamOK">
              <property name="text">
               <string>相机参数文件</string>
              </property>
              <property name="checkable">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QPushButton" name="pushButton_selectCameraParam">
           <property name="text">
            <string>相机参数文件</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QPushButton" name="pushButton_selectRawDisp">
           <property name="text">
            <string>原始视差文件</string>
           </property>
          </widget>
         </item>
         <item row="3" column="2">
          <widget class="QPushButton" name="pushButton_start">
           <property name="text">
            <string>开始测量</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_3">
        <property name="title">
         <string>显示</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <property name="topMargin">
          <number>2</number>
         </property>
         <property name="bottomMargin">
          <number>2</number>
         </property>
         <item>
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>缩放:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSlider" name="horizontalSlider_scale">
           <property name="maximum">
            <number>10</number>
           </property>
           <property name="value">
            <number>10</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_ImgSwitch">
           <property name="text">
            <string>切换画面</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>110</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>110</height>
         </size>
        </property>
        <property name="title">
         <string>结果</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <property name="topMargin">
          <number>2</number>
         </property>
         <property name="bottomMargin">
          <number>2</number>
         </property>
         <item row="1" column="1">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>X=</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QLineEdit" name="lineEdit_ImgX"/>
         </item>
         <item row="2" column="2">
          <widget class="QLineEdit" name="lineEdit_ImgY"/>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>距离:</string>
           </property>
          </widget>
         </item>
         <item row="3" column="2">
          <widget class="QLineEdit" name="lineEdit_Res"/>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>图像空间坐标:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Y=</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_4">
        <property name="title">
         <string>说明</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <property name="topMargin">
          <number>2</number>
         </property>
         <property name="bottomMargin">
          <number>2</number>
         </property>
         <item>
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>1.鼠标点击测距</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>2.[切换画面]可以在视差图和原图直接切换显示</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_8">
           <property name="text">
            <string>3.距离值的单位取决于标定时输入的标定板尺寸.</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_9">
           <property name="text">
            <string>例如使用方格边长为25mm的标定板,输入尺寸为25,
则距离的单位为mm,
如果输入2.5,则距离的单位为cm</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>33</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="0" rowspan="2">
    <widget class="QScrollArea" name="scrollArea">
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>520</width>
        <height>461</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout_4">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <property name="spacing">
        <number>0</number>
       </property>
       <item row="0" column="0">
        <layout class="QGridLayout" name="TraceLayout">
         <property name="spacing">
          <number>0</number>
         </property>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton_start</sender>
   <signal>clicked()</signal>
   <receiver>RulerView</receiver>
   <slot>onStart()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>817</x>
     <y>150</y>
    </hint>
    <hint type="destinationlabel">
     <x>265</x>
     <y>6</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_selectRawDisp</sender>
   <signal>clicked()</signal>
   <receiver>RulerView</receiver>
   <slot>onSelectRawDispFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>773</x>
     <y>46</y>
    </hint>
    <hint type="destinationlabel">
     <x>842</x>
     <y>26</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_selectOrigin</sender>
   <signal>clicked()</signal>
   <receiver>RulerView</receiver>
   <slot>onSelectOriginImg()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>748</x>
     <y>76</y>
    </hint>
    <hint type="destinationlabel">
     <x>536</x>
     <y>89</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_selectCameraParam</sender>
   <signal>clicked()</signal>
   <receiver>RulerView</receiver>
   <slot>onSelectCameraParamFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>751</x>
     <y>114</y>
    </hint>
    <hint type="destinationlabel">
     <x>537</x>
     <y>136</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_ImgSwitch</sender>
   <signal>clicked()</signal>
   <receiver>RulerView</receiver>
   <slot>onSwitchImageToShow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>817</x>
     <y>197</y>
    </hint>
    <hint type="destinationlabel">
     <x>536</x>
     <y>184</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onMouseMove(int,int)</slot>
  <slot>onMouseLButtonDown(int,int)</slot>
  <slot>onMouseRButtonDown(int,int)</slot>
  <slot>onSelectDisparityMap()</slot>
  <slot>onStart()</slot>
  <slot>onSelectRawDispFile()</slot>
  <slot>onSelectCameraParamFile()</slot>
  <slot>onSwitchImageToShow()</slot>
  <slot>onSelectOriginImg()</slot>
 </slots>
</ui>
